benches/src/main.rs

```diff
@@ -0,0 +1,235 @@
+    #[clap(long, value_delimiter = ',')]
+    versions: Option<Vec<String>>,
+
+    /// Force install Foundry versions
+    #[clap(long)]
+    force_install: bool,
+
+    /// Show verbose output
+    #[clap(long)]
+    verbose: bool,
+
+    /// Directory where the aggregated benchmark results will be written.
+    #[clap(long, default_value = ".")]
+    output_dir: PathBuf,
+
+    /// Name of the output file (default: LATEST.md)
+    #[clap(long, default_value = "LATEST.md")]
+    output_file: String,
+
+    /// Run only specific benchmarks (comma-separated:
+    /// forge_test,forge_build_no_cache,forge_build_with_cache,forge_fuzz_test,forge_coverage)
+    #[clap(long, value_delimiter = ',')]
+    benchmarks: Option<Vec<String>>,
+
+    /// Run only on specific repositories (comma-separated in org/repo[:rev] format:
+    /// ithacaxyz/account,Vectorized/solady:main,foundry-rs/foundry:v1.0.0)
+    #[clap(long, value_delimiter = ',')]
+    repos: Option<Vec<String>>,
+}
+
+/// Mutex to prevent concurrent foundryup calls
+static FOUNDRY_LOCK: Lazy<Mutex<()>> = Lazy::new(|| Mutex::new(()));
+fn switch_version_safe(version: &str) -> Result<()> {
+    let _lock = FOUNDRY_LOCK.lock().unwrap();
+    switch_foundry_version(version)
+}
+
+#[allow(unused_must_use)]
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    let cli = Cli::parse();
+
+    // Check if hyperfine is installed
+    let hyperfine_check = Command::new("hyperfine").arg("--version").output();
+    if hyperfine_check.is_err() || !hyperfine_check.unwrap().status.success() {
+        eyre::bail!(
+            "hyperfine is not installed. Please install it first: https://github.com/sharkdp/hyperfine"
+        );
+    }
+
+    // Determine versions to test
+    let versions = if let Some(v) = cli.versions {
+        v
+    } else {
+        FOUNDRY_VERSIONS.iter().map(|&s| s.to_string()).collect()
+    };
+
+    // Get repo configurations
+    let repos = if let Some(repo_specs) = cli.repos.clone() {
+        repo_specs
+            .iter()
+            .map(|spec| RepoConfig::try_from(spec.as_str()))
+            .collect::<Result<Vec<_>>>()?
+    } else {
+        BENCHMARK_REPOS.clone()
+    };
+
+    sh_println!("🚀 Foundry Benchmark Runner");
+    sh_println!("Running with versions: {}", versions.join(", "));
+    sh_println!(
+        "Running on repos: {}",
+        repos.iter().map(|r| format!("{}/{}", r.org, r.repo)).collect::<Vec<_>>().join(", ")
+    );
+
+    // Install versions if requested
+    if cli.force_install {
+        install_foundry_versions(&versions)?;
+    }
+
+    // Determine benchmarks to run
+    let benchmarks = if let Some(b) = cli.benchmarks {
+        b.into_iter().filter(|b| ALL_BENCHMARKS.contains(&b.as_str())).collect()
+    } else {
+        // Default: run all benchmarks except fuzz tests and coverage (which can be slow)
+        vec!["forge_test", "forge_build_no_cache", "forge_build_with_cache"]
```

Comment:
we could use an array here or do to_string on all items but doesnt really matter